Meta image data

arunbalajeev opened the pull request:
@Atanahel - MetaImageData class is made as a super class for storing local parameters. Presently, I tried to put only 2 parameters - Compression State(earlier defined in ImageData Class), type of Compression(String). Could you please chech the structure and content of MetaImageData class and ImageData class.

SeguinBe commented on preprocessing/src/main/java/ImageData.java:
```diff
@@ -25,8 +25,8 @@
-    public enum ImageDataState {UNCOMPRESSED, COMPRESSED, ERROR}
-    private ImageDataState state;
+    //public enum ImageDataState {UNCOMPRESSED, COMPRESSED, ERROR}
+    private main.java.MetaImageData.ImageDataState state;
```
This element exists already in the class above, does not make sense to redefine it here

arunbalajeev replied:
Variable state was not visible int ImageData class. But, I removed it from ImageData class and changed the visibility of variable state in MetaImageData class to public

SeguinBe commented:
I commented a few things on the local scale.

On the global scale, what is the underlying choice of having the metadata part the inherited part? So that you can create a new image with the same metadata?
Usually the inherited part is the core part. What do you think?

arunbalajeev commented:
I tried to rectify and make some changes as per your suggestions.
1. Changed SerialVersionID.
2. Removed the readMetaObject and writeMetaObject.
3.Removed repeated initialization of the variable "state"
On the global scale, we have ImageData inherited from the MetaImageData class. Hence, ImageData class will have all the same parameters mentioned in MetaImageData and these parameters will be common for all ImageData objects. I think we can also override over the parameters within ImageData object, if required.

arunbalajeev commented:
@Atanahel : As we discussed today, I have modified the branch MetaImageData keeping the original ImageData class and modifying MetaImageData class and also created metadata class. Could you please check it and comment on it.

arunbalajeev commented:
@Atanahel : As we discussed today, I have modified the branch MetaImageData keeping the original ImageData class and modifying MetaImageData class and also created metadata class. Could you please check it and comment on it.

SeguinBe commented:
Erf, you should have just started from a new branch to avoid all the unnecessary previous commits.

Also, you NEVER commit things by copy-pasting things from the master branch (like you did with NativeLibraryLoader), you merge the master branch to your branch if necessary and it does the job for you.
